software/main.py

```diff
@@ -0,0 +1,266 @@
+                        if(abs(basket.x-reso_x_mid)<10):
+                            robot.stop()
+                            state="throw"
+                            print("Robot is ready to throw")
+                            #state="find_ball"
+                            
+                        elif abs(basket.x-reso_x_mid)>30:
+                            #orbit_speeds = Orbit(radius, reso_x_mid, xcord, dist)
+                             
+                            speedY=0
+                            speedX=0.5
+                            speedR=1000*speedX/(640-radius)
+
+                            if radius > (dist-5):
+                                speedY += (radius-dist)/100
+                            elif radius > (dist+5):
+                                speedY += (radius-dist)/100
+                            
+                            
+                            if xcord > (reso_x_mid + 1) or xcord < (reso_x_mid - 1):
+                                speedR += (reso_x_mid- xcord) / 100 
+                                                   
+                            #robot.move(orbit_speeds[0],orbit_speeds[1],orbit_speeds[2])
+                            robot.move(speedX, speedR, speedY)                        
+                    else:
+                        #Orbit
+                        #orbit_speeds = Orbit(radius, reso_x_mid, xcord, dist)
+
+                        speedY=0
+                        speedX=0.5
+                        speedR=1000*speedX/(640-radius)
+
+                        if radius > (dist-5):
+                            speedY += (radius-dist)/100
+                        elif radius > (dist+5):
+                            speedY += (radius-dist)/100
+                        
+                        
+                        if xcord > (reso_x_mid + 1) or xcord < (reso_x_mid - 1):
+                            speedR += (reso_x_mid- xcord) / 100 
+                                                
+                        #robot.move(orbit_speeds[0],orbit_speeds[1],orbit_speeds[2])
+                        robot.move(speedX, speedR, speedY)
+                        
+                        #robot.move(orbit_speeds[0],orbit_speeds[1],orbit_speeds[2])
+            
+                            
+                    
+
+            elif state == "throw":
+                #calc speed in m/s
+
+                processedData.balls.sort(key= lambda x: x.size)
+                targeted_ball=processedData.balls[-1]
+                dist=targeted_ball.distance
+                delta=((xcord-reso_x_mid)/reso_x_mid)
+                
+
+                
+                #Moves to ball
+                speedY = 0.3
+                if dist > 400:
+                    speedR = 0
+                    speedX = 0
+                    speedT=950
+                    baskety=processedData.basket_m.y
+                    for i in range(len(tsize)):

```

Comment:
I suggest using linear interpolation to determine throwing speeds. Additionally speed calculations could be done in your calculations class

Reply:
I use two cross products and their mean.